### ISSUE TITLE
[URL Unfurling]: Update image context menu with Copy/Open Link actions

### ISSUE THREAD

**Issue by igor-sirotin** [status-im/status-desktop#11941]
# Introduction

With the new URL unfurling image links are unfurled in a different way.

Now, instead of downloading the images on the receiver side, they're:
- downloaded on the sender side
- compressed
- and attached to the sent message.

The receiver doesn't have to unfurl the link and download the image. Simply take the attached image and show it.

# Description

Currently the image context menu provides 2 options:
- Copy image/GIF
- Download image/GIF

**Because the image is compressed**, it would be nice to give the user option to get the original image.

# Acceptance Criteria

For an **unfurled** image the context menu should has these options:
- Copy Link
- Open Link

For a simple attached image/GIF the context menu should keep current options:
- Copy image/GIF
- Download image/GIF

# Design

<details>
    <summary>Discord example</summary>

<img width="931" alt="image" src="https://github.com/status-im/status-desktop/assets/25482501/83a826f6-8f25-4f6d-be5e-c2e34cb1b618">
<img width="924" alt="image" src="https://github.com/status-im/status-desktop/assets/25482501/3b5f645c-9244-41f1-b913-c8f44067f3b6">
</details>

https://www.figma.com/file/Mr3rqxxgKJ2zMQ06UAKiWL/%F0%9F%92%AC-Chat%E2%8E%9CDesktop?type=design&node-id=22376-63722&mode=design&t=Jz2QtPMMgdwxN0E2-4

<img width="529" alt="image" src="https://github.com/status-im/status-desktop/assets/25482501/199bb696-e30d-444f-bed5-6c32547b3ac5">
<img width="693" alt="image" src="https://github.com/status-im/status-desktop/assets/25482501/1f0bb7a4-ced3-44c7-a6ba-070b062019cd">

# A note about GIFs

GIFs are not yet unfurled on the sender side. So we want to keep the copy/download options for those.

**Comment by caybro** [comment]
I don't understand the motivation behind losing a feature to directly download an image, @benjthayer ?

**Comment by igor-sirotin** [comment]
@alexandraB99 @caybro, I've updated the issue description. Sorry for being unclear previously.
Also added links to design which @benjthayer finished just this week.

I know we didn't yet decide if we should keep the copy/download options in the new case of unfurled image links. But from code perspective the 90% of work will be to carefully propagate the `url` property to `ImageContextMenu`. And keeping (or removing) the original context menu options should be very easy with a simple condition. So we can start implementing it and make last minor changes in the end.

**Comment by benjthayer** [comment]
@caybro @igor-sirotin just to add some additional context incase helpful: 

The mobile designs for link previews were done first by Mario. I've then subsequently worked with Igor and Jonathan to align on desktop.

Through that process, we explored 2 routes for how we handle linked images. 1 aligns with mobile and shows linked images being sent as an unfurl "card" which can be seen here:
https://www.figma.com/file/Mr3rqxxgKJ2zMQ06UAKiWL/%F0%9F%92%AC-Chat%E2%8E%9CDesktop?type=design&node-id=22375-50744&mode=design&t=Iqp2HeF8DTYJy7Ht-4

The other aligns with Discord where the linked image is shown as an image when sent (removing the link). This route can be seen here:
https://www.figma.com/file/Mr3rqxxgKJ2zMQ06UAKiWL/%F0%9F%92%AC-Chat%E2%8E%9CDesktop?type=design&node-id=22347-216740&mode=design&t=Iqp2HeF8DTYJy7Ht-4

With the Discord route, Discord limit the contextual options to `Copy Link` and `Open Link`. But for now, I've also kept the options to `Copy image` and `Download image`. This is because the user would expect to see those options given the link is removed when the message is sent meaning there is no differentiation between an image that has been sent embedded in the message vs linked to. Igor raised the issue that the compression of linked images is inferior and for that reason we may want to remove the `Copy image` and `Download image` options. 

In summary, we need to discuss with @John-44 1) which route we should go with and 2) if going the Discord route, whether the `Copy image` and `Download image` options should be retained.
